### ISSUE TITLE
crash on trying to enter File or Edit menu (not View or Help) "Invalid borders specified for theme pixmap", Breeze theme

### ISSUE THREAD

**Issue by milkmiruku** [ahlstromcj/sequencer64#126]
```
** (seq64:25277): WARNING **: Invalid borders specified for theme pixmap:
        /usr/share/themes/Breeze/gtk-2.0/../assets/line-h.png,
borders don't fit within the image
zsh: trace trap (core dumped)  seq64
```
```
Sequencer64 0.94.1
Native JACK/ALSA (rtmidi) * = ON
ALSA-only MIDI support * = off
PortMIDI support * = off
Event editor * = ON
Event multimap (vs list) = off
Follow progress bar = ON
Highlight edit pattern * = ON
Highlight empty patterns = ON
JACK session * = ON
JACK support * = ON
LASH support * = off
MIDI vector (vs list) = ON
Pause support * = ON
Save time-sig/tempo = ON
Seq32 chord generator * = ON
Seq32 LFO support * = ON
Seq32 menu buttons = ON
Seq32 transpose * = ON
Solid piano-roll grid = ON
Main window scroll-bars = off
Multiple main windows * = ON
Box song selection = off
Statistics support * = off
Windows support * = off
Debug code * = ON
64-bit support enabled
```

I'll try another theme shortly but AFAIK Breeze is isn't that relatively uncommon a theme, or at least a crash might possibly be a rash action :)

**Comment by milkmiruku** [comment]
It doesn't crash using the Adapta theme. Still no numbers though.

**Comment by ahlstromcj** [comment]
I installed the Breeze GTK theme on this laptop, and it does replicate the crash.  Pretty weird!!  I suspect it's an issue with the theme itself, but I will look into further later today.  Thanks!

**Comment by ahlstromcj** [comment]
After some investigration, I believe this is a Breeze theme issue, where the border setting for the menu separator line doesn't work with the 20x1 pixel line-h.png file.  I couldn't figure out how to fix the theme locally.  The workaround is to comment out the code in mainwnd that adds the SeparatorElem (horizontal lines) in the menu.  I might provide a macro for that, but fixing the theme would be better.

**Comment by ahlstromcj** [comment]
Okay, I found an edit to the theme that prevents the crash.   However, instead of a line separator in menus that have a separator, you will see just some extra vertical space.  To implement the workaround:

As root, edit /usr/share/themes/gtk-2.0/Breeze/gtkrc (or gtk-3.0/gtkrc?).  Find the second occurrence of "../assets/line-h.png", in the "separator_menu_item" section, and comment out the "file" and "border" lines.

You might try playing with editing the PNG file and tweaking the border sections, if you're keen to dig into Gtkrc settings :-D

**Comment by ahlstromcj** [comment]
I ran this scenario through valgrind, and the segfault (SIGTRAP) stack trace shows no code from seq64.  I'm tempted to close it.  @milkmiruku what do you mean by "still no numbers though"?  Thanks!

**Comment by milkmiruku** [comment]
@ahlstromcj that was just my misinterpretation of the config, but this issue is resolved with the latest tempo_track build :)

**Comment by milkmiruku** [comment]
Oops, I spoke too soon;

```
[JACK server not running?]
[JACK server not running, JACK sync disabled]
[JACK sync disabled]
==27149== Conditional jump or move depends on uninitialised value(s)
==27149==    at 0x10B21C90: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B22B17: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B22BC1: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B1F522: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x79159E0: ??? (in /usr/lib/libgtk-x11-2.0.so.0.2400.32)
==27149==    by 0x78AE7CB: ??? (in /usr/lib/libgtk-x11-2.0.so.0.2400.32)
==27149==    by 0x9208B71: g_closure_invoke (in /usr/lib/libgobject-2.0.so.0.5400.3)
==27149==    by 0x921C5A9: ??? (in /usr/lib/libgobject-2.0.so.0.5400.3)
==27149==    by 0x9225190: g_signal_emit_valist (in /usr/lib/libgobject-2.0.so.0.5400.3)
==27149==    by 0x922623F: g_signal_emit (in /usr/lib/libgobject-2.0.so.0.5400.3)
==27149==    by 0x79C9234: ??? (in /usr/lib/libgtk-x11-2.0.so.0.2400.32)
==27149==    by 0x7832822: gtk_container_propagate_expose (in /usr/lib/libgtk-x11-2.0.so.0.2400.32)
==27149== 

** (seq64:27149): WARNING **: Invalid borders specified for theme pixmap:
        /usr/share/themes/Breeze/gtk-2.0/../assets/line-h.png,
borders don't fit within the image
==27149== 
==27149== Process terminating with default action of signal 5 (SIGTRAP): dumping core
==27149==    at 0x949E982: ??? (in /usr/lib/libglib-2.0.so.0.5400.3)
==27149==    by 0x949FCEC: g_logv (in /usr/lib/libglib-2.0.so.0.5400.3)
==27149==    by 0x949FE4F: g_log (in /usr/lib/libglib-2.0.so.0.5400.3)
==27149==    by 0x10B21B30: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B22B17: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B22BC1: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B1F8F7: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x10B209EA: ??? (in /usr/lib/gtk-2.0/2.10.0/engines/libpixmap.so)
==27149==    by 0x78BD45F: ??? (in /usr/lib/libgtk-x11-2.0.so.0.2400.32)
==27149==    by 0x78AE7CB: ??? (in /usr/lib/libgtk-x11-2.0.so.0.2400.32)
==27149==    by 0x9208B71: g_closure_invoke (in /usr/lib/libgobject-2.0.so.0.5400.3)
==27149==    by 0x921C5A9: ??? (in /usr/lib/libgobject-2.0.so.0.5400.3)
==27149== 
==27149== HEAP SUMMARY:
==27149==     in use at exit: 4,826,372 bytes in 36,266 blocks
==27149==   total heap usage: 158,608 allocs, 122,342 frees, 23,607,520 bytes allocated
==27149== 
==27149== LEAK SUMMARY:
==27149==    definitely lost: 5,120 bytes in 11 blocks
==27149==    indirectly lost: 23,517 bytes in 1,003 blocks
==27149==      possibly lost: 50,825 bytes in 1,423 blocks
==27149==    still reachable: 4,429,958 bytes in 31,809 blocks
==27149==                       of which reachable via heuristic:
==27149==                         length64           : 27,104 bytes in 365 blocks
==27149==                         newarray           : 40,832 bytes in 78 blocks
==27149==                         multipleinheritance: 32,800 bytes in 418 blocks
==27149==         suppressed: 0 bytes in 0 blocks
==27149== Rerun with --leak-check=full to see details of leaked memory
==27149== 
==27149== For counts of detected and suppressed errors, rerun with: -v
==27149== Use --track-origins=yes to see where uninitialised values come from
==27149== ERROR SUMMARY: 20 errors from 1 contexts (suppressed: 5 from 1)
zsh: trace trap (core dumped)  valgrind seq64
```

**Comment by mrlinuxfish** [comment]
I am also having a similar problem. Editing the breeze theme does not seem like a reasonable solution to this issue. Whenever a new update to the breeze theme comes out it will probably cause issues again. If this is a problem with breeze, the issue should be sent upstream for resolution.

**Comment by ahlstromcj** [comment]
Unfortunately, the only thing we can do on seq64's end is add a krufty option to not add separators to seq64 menus.

Another option is to use the Qt version of seq64 instead.  That will ultimately become the main version, but it currently still has a few issue that will be fixed in a few more revisions of the 0.96 series in the qt5_reconcile branch.

I think somebody said they'd reported the breeze issue upstream.  For me, it is easier to avoid that theme, nice though it looks.  There's also a freeze-up issue with Gtk when using hot-keys for the menu before clicking in the GUI.  These issues are part of the reason why we're moving to Qt.
